frappe/core/doctype/file/utils.py

```diff
@@ -357,6 +358,49 @@ def attach_files_to_document(doc: "Document", event) -> None:
 			doc.log_error("Error Attaching File")
 
 
+def relink_files(doc, fieldname, temp_doc_name):
+	from frappe.utils.data import add_to_date, now_datetime
+
+	"""
+	Relink files attached to incorrect document name to the new document name
+	by check if file with temp name exists that was created in last 60 minutes
+	"""
+	mislinked_file = frappe.db.exists(
+		"File",
+		{
+			"file_url": doc.get(fieldname),
+			"attached_to_name": temp_doc_name,
```

Comment:
Check if temp_doc_name is truthy? if it's falsy we shouldn't attempt linking.

Reply:
relink_files is not called from relink_mismatched_files if temp_doc_name is false. i can also add it here if required.